lib/request.js

```diff
@@ -60,9 +88,20 @@ class Request {
 
     if (typeof cb !== 'function') return request(requestOptions);
 
+    const auth = this.authProvider;
     return request(requestOptions, (err, res, body) => {
       if (err) return cb(err);
-      cb(null, { statusCode: res.statusCode, body: body });
+
+      // Refresh auth if 401
+      if (res.statusCode === 401 && auth.type) {
+        return refreshAuth(auth.type, auth.config)
+          .then(newAuth => {
+            this.requestOptions.auth = newAuth;
+            return this.request(method, options, cb);
```

Comment:
Can we call `request` directly (instead of the recursive call)?  I worry about the case where the HTTP request keeps 401ing for some reason, and we end up in an infinite request loop / oom situation.